invenio_banners/resources/resource.py

```diff
@@ -30,41 +34,99 @@ def create_url_rules(self):
         routes = self.config.routes
         return [
             route("POST", routes["create"], self.create),
+            route("GET", routes["banner"], self.read),
+            route("GET", routes["list"], self.search),
+            route("DELETE", routes["banner"], self.delete),
+            route("GET", routes["active"], self.read_active),
+            route("PUT", routes["banner"], self.update),
         ]
 
-    #
-    # Primary Interface
-    #
+    @request_view_args
+    @request_data
+    @response_handler()
+    def update(self):
+        """Update a banner."""
+        banner_id = resource_requestctx.view_args["banner_id"]
+        banner = self.service.update(
+            id=banner_id,
+            identity=g.identity,
+            data=resource_requestctx.data,
+        )
+
+        if banner is None:
+            abort(
+                404, {"message": "Banner with id {0} is not found.".format(banner_id)}
+            )
```

Comment:
Ideally the errors should be handled with exceptions.
In the service layer we should raise an exception and then we create an errorHandler class (see [example](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/resources/errors.py#L97) in invenio-records-resources) that we pass to the resource class (see example [here](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/resources/records/resource.py#L49)). 
By doing so we can get rid of this duplicated code and manage errors code easier.